src/database/DataTypes/StocktakeBatch.js

```diff
@@ -211,7 +211,25 @@ export class StocktakeBatch extends Realm.Object {
    */
   set countedTotalQuantity(quantity) {
     // Handle packsize of 0.
-    this.countedNumberOfPacks = this.packSize ? quantity / this.packSize : 0;
+    const countedNumberOfPacks = this.packSize ? quantity / this.packSize : 0;
+
+    this.countedNumberOfPacks = countedNumberOfPacks;
+    this.doses = countedNumberOfPacks * this.dosesPerVial;
```

Comment:
Should this possibly be `countedNumberOfDoses`?

Reply:
that's probably a better name hmmmm